### PR TITLE
Added tests

### PR DESCRIPTION
**Pull request by Annajoe96**
| <a href="https://ellipsis.dev" target="_blank">Ellipsis</a> | :rocket: This PR description was created by [Ellipsis](https://www.ellipsis.dev) for commit 6ff56bde7e418fed6b9ad2e7e3366dc24f302f8d.  | 
|--------|--------|

### Summary:
This PR adds comprehensive unit and integration tests for the `CheckoutSummary` class in the `calculator.rb` file, covering various scenarios for adding and removing discounts and items, as well as calculating the total price.

**Key points**:
- Added `calculator_tests.rb` file with unit and integration tests for `CheckoutSummary` class.
- Tests cover adding and removing discounts and items.
- Tests also cover calculating the total price with various combinations of discounts and items.
- Edge cases for invalid inputs are also tested.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: WIP/calculator_tests.rb [hunk 1]
```diff
@@ -0,0 +1,376 @@
+require './calculator.rb'
+
+
+class Calculatortest < Minitest::Test
```
Comment (danielpaul) [inline WIP/calculator_tests.rb]: camel-cased class names

File: WIP/calculator_tests.rb [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+
+# --------------------------------------------------
+# Test for adding discounts
+# --------------------------------------------------
+
+  def test_add_discount
+    @price = Calculate.new
+
+    result = @price.add_discount({
+    name: "new yeardeal",
```
Comment (danielpaul) [inline WIP/calculator_tests.rb]: indentation

File: WIP/calculator_tests.rb [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+    amount: nil,
+    percentage: 20
+
+    })
+
+    # result = @price.return_discounts.size
```
Comment (danielpaul) [inline WIP/calculator_tests.rb]:
```suggestion
```

File: WIP/calculator_tests.rb [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+
+    assert result == true
+  end
+
+  def test_add_two_discounts
+    @price = Calculate.new
+
+    @price.add_discount({
+    name: "new yeardeal",
```
Comment (danielpaul) [inline WIP/calculator_tests.rb]: indentations on all add_discount methods

**Review by danielpaul — CHANGES_REQUESTED** [review]
